Tests/MongoSwiftSyncTests/TransactionsTests.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+import MongoSwift
+import Nimble
+import TestsCommon
+
+/// Struct representing a single test within a spec test JSON file.
+private struct TransactionsTest: SpecTest {
+    let description: String
+
+    let operations: [TestOperationDescription]
+
+    let outcome: TestOutcome?
+
+    let skipReason: String?
+
+    let useMultipleMongoses: Bool?
+
+    let clientOptions: TestClientOptions?
+
+    let failPoint: FailPoint?
+
+    let sessionOptions: [String: ClientSessionOptions]?
+
+    let expectations: [TestCommandStartedEvent]?
+
+    static let sessionNames: [String] = ["session0", "session1"]
+}
+
+/// Struct representing a single transactions spec test JSON file.
+private struct TransactionsTestFile: Decodable, SpecTestFile {
+    private enum CodingKeys: String, CodingKey {
+        case name, runOn, databaseName = "database_name", collectionName = "collection_name", data, tests
+    }
+
+    let name: String
+
+    let runOn: [TestRequirement]?
+
+    let databaseName: String
+
+    let collectionName: String?
+
+    let data: TestData
+
+    let tests: [TransactionsTest]
+}
+
+final class TransactionsTests: MongoSwiftTestCase, FailPointConfigured {
+    var activeFailPoint: FailPoint?
+
+    override func tearDown() {
+        self.disableActiveFailPoint()
+    }
+
+    override func setUp() {
+        self.continueAfterFailure = false
+    }
+
+    func testTransactions() throws {
+        let skippedTestKeywords = [
+            "count", // skipped in RetryableReadsTests.swift
```

Comment:
so the reason we skip this in retryable reads is that the tests are written using the old count API, which was last minute deprecated right before the 4.0 release. we don't have the old API at all but some older drivers that haven't tagged a major version since then still do.

in this case, it looks like the txns count tests are checking that certain errors occur when you try to run `count` in a transactions, since that isn't supported by the server. since we don't provide an API for running the `count` command I think it's ok to skip this, but we could update the comment to be more descriptive.

we could consider converting it to `runCommand` and running count that way since a user could technically still do it, but that seems probably like overkill.

Reply:
Ok, I've updated the comment to be more specific about why `count` is implemented.

Reply:
So we actually do call `count` under the hood in `estimatedDocumentCount`. That being said, [`estimatedDocumentCount` is forbidden from accepting a session by the sessions spec](https://github.com/mongodb/specifications/blob/master/source/sessions/driver-sessions.rst#new-collection-methods-that-take-an-explicit-session) because `count` isn't supported in transactions, so it's a little puzzling why this tests exists in the first place.

So for that reason, I think we should continue to skip the test, but we should remove the session parameter from `estimatedDocumentCount`, and also perhaps add a line to its docstring mentioning that it cannot be used in a transaction.

Reply:
as for why the test exists - it just checks that you properly error when a user tries to call `count` in a transaction.

Reply:
ok I see your edit now jinx lol

Reply:
Ok, I've removed the session parameter from `estimatedDocumentCount` and added a line to the docstring mentioning that it can't be used in a transaction.